### PR TITLE
[WIP] Hamburger menu

### PR DESCRIPTION
**Pull request by ragesoss**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -92,6 +92,7 @@
     "react-addons-css-transition-group": "^15.4.1",
     "react-addons-shallow-compare": "^15.4.1",
     "react-addons-test-utils": "^15.4.1",
+    "react-burger-menu": "^2.1.4",
```
Comment (ragesoss) [inline package.json]: Did you add this manually, or with `yarn add`? If you do it with yarn, it should also update the yarn.lock file.

Comment (sejalkhatri) [inline package.json]: Added using this command: npm install react-burger-menu --save

Comment (ragesoss) [inline package.json]: `yarn add` is now the way to do it, to ensure that we have the version locking features that yarn provides.

File: app/assets/javascripts/components/nav.jsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  updateDimensions() {
+    this.setState({ width: $(window).width(), height: $(window).height() });
+  },
+
+  showSettings(event) {
+    event.preventDefault();
+  },
+
+  render() {
+    let navBar;
```
Comment (ragesoss) [inline app/assets/javascripts/components/nav.jsx]: This is going to be a pretty complex bunch of logic to handle all the different switches of possible links, but they are mostly independent of one another.

Whitespace, comments, and grouping of related code into isolated units will be really helpful to keep this easy to work with and refactor. (See `CourseNavbar` for a similar situation.)

File: app/assets/javascripts/components/nav.jsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        </span>
+      );
+    }
+    if (this.state.fluid)
+    {
+      navClass = "top-nav fluid";
+    } else {
+      navClass = "top-nav";
+    }
+    if (this.state.width < 500)
```
Comment (ragesoss) [inline app/assets/javascripts/components/nav.jsx]: Maybe we should factor these if/else branches out into two separate presentational components, one for mobile and one for desktop?

File: app/assets/javascripts/components/nav.jsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    const fluid = $('#nav_root').data('fluid');
+    const exploreurl = $('#nav_root').data('exploreurl');
+    const explorename = $('#nav_root').data('explorename');
+    const exploreclass = $('#nav_root').data('exploreclass');
+    const usersignedin = $('#nav_root').data('usersignedin');
+    const ifadmin = $('#nav_root').data('ifadmin');
+    const trainingurl = $('#nav_root').data('trainingurl');
+    const disableTraining = $('#nav_root').data('disable_training');
+    const helpDisabled = $('#nav_root').data('help_disabled');
+    const askUrl = $('#nav_root').data('ask_url');
```
Comment (ragesoss) [inline app/assets/javascripts/components/nav.jsx]: The 'ask a question' form from the haml version isn't implemented yet.

Comment (sejalkhatri) [inline app/assets/javascripts/components/nav.jsx]: @ragesoss i have doubts related to this, about how to send data to a new window opened from react.

File: app/assets/javascripts/components/nav.jsx (hunk 2 above)
Comment (sejalkhatri) [inline app/assets/javascripts/components/nav.jsx]: I'll refer it, thanks!

File: app/assets/javascripts/components/nav.jsx [hunk 5]
```diff
@@ -66,6 +68,15 @@ const Nav = React.createClass({
...
     event.preventDefault();
   },
+  handleQueryChange(event) {
+    console.log(this.state.query);
+    this.setState({ query: event.target.value });
+  },
+  handleSubmit() {
+    // const query = this.state.query.trim();
+
+    window.open("/ask");
```
Comment (sejalkhatri) [inline app/assets/javascripts/components/nav.jsx]: here! I have to send the query data to the new window.

File: app/views/layouts/admin.html.haml [hunk 6]
```diff
@@ -11,7 +11,7 @@
   %body{class: body_class(request)}
     = render "shared/flash"
     %div.wrapper
-      = render "shared/nav"
+      #nav_root{"data-rooturl" => main_app.root_url, "data-logopath" => logo_path, "data-exploreclass" => 'inactive', "data-fluid" => "false", "data-exploreurl" =>  main_app.explore_path, "data-explorename" => t(Features.wiki_ed? ? "application.explore" : "courses_generic.explore"), "data-usersignedin" => user_signed_in? ? 'true' : 'false', "data-ifadmin" => current_user&.admin? ? 'true' :'false', "data-trainingurl": main_app.training_url, "data-disable_training": Features.disable_training? ? 'true' : 'false', "data-help_disabled" => Features.disable_help? ? 'true' : 'false', "data-ask_url" => main_app.ask_url, "data-user_permissions" => permissions, "data-wiki_ed" => Features.wiki_ed? ? 'true' : 'false', "data-language_switcher_enabled" => languageswitcherenabled, "data-username" => current_user&.username, "data-destroyurl" => main_app.destroy_user_session_url, "data-omniauth_url" => main_app.user_mediawiki_omniauth_authorize_url}
```
Comment (sejalkhatri) [inline app/views/layouts/admin.html.haml]: @ragesoss I am having problem connecting the react nav_root to the admin rails view.

Comment (ragesoss) [inline app/views/layouts/admin.html.haml]: can you elaborate?

Comment (ragesoss) [inline app/views/layouts/admin.html.haml]: The main.<hash>.js file is not loaded in the admin layout. You'll need to add it. That may be a problem with other layouts as well.

**Comment by sejalkhatri** [comment]
The problem with delayed loading of Navbar Js component is resolved by running gulp build.

**Pull request closed without merging**